### PR TITLE
add drops to turrets/nonmilitary robots

### PR DESCRIPTION
**Pull request by misterprimus**
#### Summary
``SUMMARY: Features "Make secubots/turrets drop ammo and a salvageable robot carcass."``

#### Purpose of change
Balance/realism

#### Describe the solution
Mostly changes secubots/turrets from “explode” to “broken.” Doesn’t make sense for turrets/civilian bots to come with self-destruct features like a tank or for rifle/pistol ammo to cook off from a shot. IRL a round would penetrate a power supply or circuit board and the robot would just shut down.
The biggest changes are the addition of a “drops_ammo” tag and the changes of the carcass generation script to check for that tag and then perform a loop to check for and generate the ammo drops alongside the carcass.

#### Describe alternatives you've considered
I considered putting the ammo into the carcass of the bot, to be disassembled later, but I don’t think there is any infrastructure for this. Any inactive bot only inherits amount of damage, not any specifics like unspent ammo, so it would be possible to recover 9mm rounds from an empty turret, which would have made no sense.

#### Additional context
You guys must have been messing around with STUN_IMMUNE, so I had to put it back into my code to make it compile, but omitted it from the stuff I am posting to make my changes more clear. Also some stuff with ceramic plates for dragon armor? Anyway, I didn't test every possible combination of enemy, but it seems to be working fine since the code is not really item-specific but pretty general and makes liberal use of tags/existing infrastructure.

### CODE REVIEW

File: data/json/items/generic.json [hunk 1]
```diff
@@ -2893,5 +2893,92 @@
     "material": [ "steel", "plastic", "wood" ],
     "weight": 11339,
     "volume": 12
+  },
+
```
Comment (anothersimulacrum) [inline data/json/items/generic.json]:
```suggestion
```

File: data/json/items/generic.json [hunk 2]
```diff
@@ -2893,5 +2893,92 @@
...
+    {
```
Comment (anothersimulacrum) [inline data/json/items/generic.json]:
```suggestion
  {
```

File: data/json/items/generic.json [hunk 3]
```diff
@@ -2893,5 +2893,92 @@
...
+    "price": 1000,
+    "material": [ "steel", "plastic" ],
+    "weight": 62650,
+    "volume": 120,
+    "bashing": 4,
+    "cutting": 4,
+    "to_hit": -3,
+    "flags": [ "TRADER_AVOID", "NO_REPAIR" ]
+  },
+
```
Comment (anothersimulacrum) [inline data/json/items/generic.json]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 4]
```diff
@@ -399,6 +399,33 @@
...
+      [ [ "omni_wheel", 1 ] ],
+      [ [ "copbot_chassis", 1 ] ],
+      [ [ "android_arms", 1 ] ],
+      [ [ "power_supply", 4 ] ],
+      [ [ "solar_cell", 2 ] ],
+	  [ [ "hk_mp5", 1 ] ],
+      [ [ "canister_empty", 1 ] ]
+    ]
+  },  
+
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 5]
```diff
@@ -399,6 +399,33 @@
...
+      [ [ "android_arms", 1 ] ],
+      [ [ "power_supply", 4 ] ],
+      [ [ "solar_cell", 2 ] ],
+	  [ [ "hk_mp5", 1 ] ],
+      [ [ "canister_empty", 1 ] ]
+    ]
+  },  
+  
+  
+
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 6]
```diff
@@ -399,6 +399,33 @@
...
+      [ [ "copbot_chassis", 1 ] ],
+      [ [ "android_arms", 1 ] ],
+      [ [ "power_supply", 4 ] ],
+      [ [ "solar_cell", 2 ] ],
+	  [ [ "hk_mp5", 1 ] ],
+      [ [ "canister_empty", 1 ] ]
+    ]
+  },  
+  
+
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 7]
```diff
@@ -421,6 +448,90 @@
       [ [ "scrap", 4 ] ]
     ]
   },
+
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 8]
```diff
@@ -421,6 +448,90 @@
       [ [ "scrap", 4 ] ]
     ]
   },
+  
+    {
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
  {
```

File: data/json/recipes/recipe_deconstruction.json [hunk 9]
```diff
@@ -421,6 +448,90 @@
...
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "hk_mp5", 2 ] ],
+      [ [ "scrap", 4 ] ]
+    ]
+  },
+
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 10]
```diff
@@ -421,6 +448,90 @@
...
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "hk_mp5", 2 ] ],
+      [ [ "scrap", 4 ] ]
+    ]
+  },
+  
+    {
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
  {
```

File: data/json/recipes/recipe_deconstruction.json [hunk 11]
```diff
@@ -421,6 +448,90 @@
...
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "m4a1", 1 ] ],
+      [ [ "scrap", 4 ] ]
+    ]
+  },
+    {
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
  {
```

File: data/json/recipes/recipe_deconstruction.json [hunk 12]
```diff
@@ -421,6 +448,90 @@
...
+    "qualities": [ { "id": "SCREW", "level": 1 } ],
+    "components": [
+      [ [ "ai_module", 1 ] ],
+      [ [ "RAM", 1 ] ],
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "m4a1", 1 ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
    [ [ "m4a1", 1 ] ],
```

File: data/json/recipes/recipe_deconstruction.json [hunk 13]
```diff
@@ -421,6 +448,90 @@
...
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "m2browning_sawn", 1 ] ],
+      [ [ "scrap", 4 ] ]
+    ]
+  },
+    {
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
  {
```

File: src/mondeath.cpp [hunk 14]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
     }
     // make "broken_manhack", or "broken_eyebot", ...
     item_id.insert( 0, "broken_" );
-
```
Comment (anothersimulacrum) [inline src/mondeath.cpp]: Why'd you remove this line?

File: src/mondeath.cpp [hunk 15]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
     const int max_hp = std::max( z.get_hp_max(), 1 );
     const float overflow_damage = std::max( -z.get_hp(), 0 );
     const float corpse_damage = 2.5 * overflow_damage / max_hp;
     broken_mon.set_damage( static_cast<int>( std::floor( corpse_damage * itype::damage_scale ) ) );
 
     g->m.add_item_or_charges( z.pos(), broken_mon );
-
+	
+	
+//adds ammo drop
```
Comment (anothersimulacrum) [inline src/mondeath.cpp]:
```suggestion
    //adds ammo drop	
```

File: src/mondeath.cpp [hunk 16]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
     const float overflow_damage = std::max( -z.get_hp(), 0 );
     const float corpse_damage = 2.5 * overflow_damage / max_hp;
     broken_mon.set_damage( static_cast<int>( std::floor( corpse_damage * itype::damage_scale ) ) );
 
     g->m.add_item_or_charges( z.pos(), broken_mon );
-
+	
+	
+//adds ammo drop	
+    const bool dropsammo = z.type->has_flag( MF_DROPS_AMMO );
```
Comment (anothersimulacrum) [inline src/mondeath.cpp]: This is a little bit more readable
```suggestion
    const bool drops_ammo = z.type->has_flag( MF_DROPS_AMMO ); 
```

File: src/mondeath.cpp [hunk 17]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
     const float corpse_damage = 2.5 * overflow_damage / max_hp;
     broken_mon.set_damage( static_cast<int>( std::floor( corpse_damage * itype::damage_scale ) ) );
 
     g->m.add_item_or_charges( z.pos(), broken_mon );
-
+	
+	
+//adds ammo drop	
+    const bool dropsammo = z.type->has_flag( MF_DROPS_AMMO ); 
+    if( dropsammo ) {
```
Comment (anothersimulacrum) [inline src/mondeath.cpp]:
```suggestion
    if( drops_ammo ) {
```

File: src/mondeath.cpp [hunk 18]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
+    for( const auto &ammo_entry : z.type->starting_ammo ) {
```
Comment (anothersimulacrum) [inline src/mondeath.cpp]: Please replace auto with the type

File: src/mondeath.cpp [hunk 19]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
+		if (z.ammo[ammo_entry.first] > 0) {
```
Comment (anothersimulacrum) [inline src/mondeath.cpp]:
```suggestion
		if( z.ammo[ammo_entry.first] > 0 ) {
```

File: data/json/items/generic.json (hunk 1 above)
Comment (NickGrease) [inline data/json/items/generic.json]:
```suggestion
```

File: src/mondeath.cpp [hunk 20]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
+			g->m.spawn_item( z.pos(), ammo_entry.first.c_str(), z.ammo[ammo_entry.first.c_str()], 1, calendar::turn );
+		}
+	}
+    }
+//end adds ammo drop
```
Comment (anothersimulacrum) [inline src/mondeath.cpp]:
```suggestion
    //end adds ammo drop
```

File: src/monstergenerator.cpp [hunk 21]
```diff
@@ -147,7 +147,9 @@ const std::map<std::string, m_flag> flag_map = {
     { "DRIPS_NAPALM", MF_DRIPS_NAPALM },
     { "DRIPS_GASOLINE", MF_DRIPS_GASOLINE },
     { "ELECTRIC_FIELD", MF_ELECTRIC_FIELD },
-    { "LOUDMOVES", MF_LOUDMOVES }
+    { "LOUDMOVES", MF_LOUDMOVES },
+	{ "DROPS_AMMO", MF_DROPS_AMMO } //NEW
```
Comment (anothersimulacrum) [inline src/monstergenerator.cpp]:
```suggestion
    { "DROPS_AMMO", MF_DROPS_AMMO } //NEW
```

File: src/mtype.h [hunk 22]
```diff
@@ -164,7 +164,9 @@ enum m_flag : int {
     MF_ELECTRIC_FIELD,      // This monster is surrounded by an electrical field that ignites flammable liquids near it
     MF_LOUDMOVES,           // This monster makes move noises as if ~2 sizes louder, even if flying.
     MF_CAN_OPEN_DOORS,      // This monster can open doors.
-    MF_MAX                  // Sets the length of the flags - obviously must be LAST
+	MF_DROPS_AMMO,			// DROPS AMMO ON DEATH NEW!
```
Comment (anothersimulacrum) [inline src/mtype.h]:
```suggestion
    MF_DROPS_AMMO,			// DROPS AMMO ON DEATH NEW!
```

File: data/json/items/generic.json [hunk 23]
```diff
@@ -2893,5 +2893,92 @@
...
+    {
+    "type": "GENERIC",
+    "id": "broken_turret",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it lies limp on solid ground.  Could be gutted for parts.",
```
Comment (NickGrease) [inline data/json/items/generic.json]:
```suggestion
    "description": "A broken turret.  Much less threatening now that it lays limp on solid ground.  Could be gutted for parts.",
```

File: data/json/items/generic.json [hunk 24]
```diff
@@ -2893,5 +2893,92 @@
...
+    "flags": [ "TRADER_AVOID", "NO_REPAIR" ]
+  },
+  {
+    "type": "GENERIC",
+    "id": "broken_turret_rifle",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken milspec turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it lies limp on solid ground.  Could be gutted for parts.",
```
Comment (NickGrease) [inline data/json/items/generic.json]:
```suggestion
    "description": "A broken turret.  Much less threatening now that it lays limp on solid ground.  Could be gutted for parts.",
```

File: data/json/items/generic.json [hunk 25]
```diff
@@ -2893,5 +2893,92 @@
...
+    "flags": [ "TRADER_AVOID", "NO_REPAIR" ]
+  },
+  {
+    "type": "GENERIC",
+    "id": "broken_turret_bmg",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken antimateriel turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it lies limp on solid ground.  Could be gutted for parts.",
```
Comment (NickGrease) [inline data/json/items/generic.json]:
```suggestion
    "description": "A broken turret.  Much less threatening now that it's laid limp on solid ground.  Could be gutted for parts.",
```

File: data/json/items/generic.json [hunk 26]
```diff
@@ -2893,5 +2893,92 @@
...
+    "flags": [ "TRADER_AVOID", "NO_REPAIR" ]
+  },
+  {
+    "type": "GENERIC",
+    "id": "broken_laserturret",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken laser turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it lies limp on solid ground.  Could be gutted for parts.",
```
Comment (NickGrease) [inline data/json/items/generic.json]:
```suggestion
    "description": "A broken turret.  Much less threatening now that it's laid limp on solid ground.  Could be gutted for parts.",
```

File: src/mondeath.cpp [hunk 27]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
-
     item broken_mon( item_id, calendar::turn );
     const int max_hp = std::max( z.get_hp_max(), 1 );
     const float overflow_damage = std::max( -z.get_hp(), 0 );
     const float corpse_damage = 2.5 * overflow_damage / max_hp;
     broken_mon.set_damage( static_cast<int>( std::floor( corpse_damage * itype::damage_scale ) ) );
 
     g->m.add_item_or_charges( z.pos(), broken_mon );
-
+
```
Comment (NickGrease) [inline src/mondeath.cpp]: Not sure about these new lines (one maybe, two, don't think so.)

File: src/mondeath.cpp [hunk 28]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
+	
+//adds ammo drop	
+    const bool dropsammo = z.type->has_flag( MF_DROPS_AMMO ); 
+    if( dropsammo ) {
+    for( const auto &ammo_entry : z.type->starting_ammo ) {
+		if (z.ammo[ammo_entry.first] > 0) {
+			g->m.spawn_item( z.pos(), ammo_entry.first.c_str(), z.ammo[ammo_entry.first.c_str()], 1, calendar::turn );
+		}
+	}
+    }
```
Comment (NickGrease) [inline src/mondeath.cpp]:
```suggestion
}
```

File: data/json/recipes/recipe_deconstruction.json [hunk 29]
```diff
@@ -399,6 +399,31 @@
...
+      [ [ "sensor_module", 1 ] ],
+      [ [ "memory_module", 1 ] ],
+      [ [ "pathfinding_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "omni_wheel", 1 ] ],
+      [ [ "copbot_chassis", 1 ] ],
+      [ [ "android_arms", 1 ] ],
+      [ [ "power_supply", 4 ] ],
+      [ [ "solar_cell", 2 ] ],
+	  [ [ "hk_mp5", 1 ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
     [ [ "hk_mp5", 1 ] ],
     
```

File: data/json/recipes/recipe_deconstruction.json [hunk 30]
```diff
@@ -421,6 +446,88 @@
...
+    "qualities": [ { "id": "SCREW", "level": 1 } ],
+    "components": [
+      [ [ "ai_module", 1 ] ],
+      [ [ "RAM", 1 ] ],
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "hk_mp5", 2 ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
     [ [ "hk_mp5", 2 ] ],
```

File: data/json/recipes/recipe_deconstruction.json [hunk 31]
```diff
@@ -421,6 +446,88 @@
...
+    "qualities": [ { "id": "SCREW", "level": 1 } ],
+    "components": [
+      [ [ "ai_module", 1 ] ],
+      [ [ "RAM", 1 ] ],
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+    [ [ "m4a1", 1 ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
      [ [ "m4a1", 1 ] ],
```

File: data/json/recipes/recipe_deconstruction.json [hunk 32]
```diff
@@ -421,6 +446,88 @@
...
+    "qualities": [ { "id": "SCREW", "level": 1 } ],
+    "components": [
+      [ [ "ai_module", 1 ] ],
+      [ [ "RAM", 1 ] ],
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "m2browning_sawn", 1 ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
      [ [ "m2browning_sawn", 1 ] ],
```

File: data/json/recipes/recipe_deconstruction.json [hunk 33]
```diff
@@ -421,6 +446,88 @@
...
+    "qualities": [ { "id": "SCREW", "level": 1 } ],
+    "components": [
+      [ [ "ai_module", 1 ] ],
+      [ [ "RAM", 1 ] ],
+      [ [ "small_storage_battery", 1 ] ],
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+	  [ [ "laser_rifle", 1 ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
      [ [ "laser_rifle", 1 ] ],
```

File: data/json/recipes/recipe_deconstruction.json [hunk 34]
```diff
@@ -421,6 +446,88 @@
...
+	  [ [ "heavy_battery_cell", 1 ] ],
```
Comment (anothersimulacrum) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
      [ [ "heavy_battery_cell", 1 ] ],
```

File: src/mondeath.cpp (hunk 14 above)
Comment (misterprimus) [reply src/mondeath.cpp]: You mean the carriage return? Maybe a nervous tick. It's my first significant change (as opposed to balancing json) so I forgot to pay attention to new lines. Sometimes I added/deleted a bunch just to help me see where my code was.

File: src/mondeath.cpp (hunk 18 above)
Comment (misterprimus) [reply src/mondeath.cpp]: I guess the type is pair? I'll test out and see if it works.

**Comment by NickGrease** [comment]
All references to volume, as I've just recently learned, should be strings in the form/conversion of 1 == 250 ml - so turrets ("volume": 120) should be `"volume": "30 L"` @anothersimulacrum can correct me if I'm wrong, but if 'xxxx ml' evenly divides by 1000, then use `L`, otherwise use `ml` (eg. "2500 ml" as opposed to "2.5 L")

**Comment by Fris0uman** [comment]
Decontruction recipe for vroken robot should be the same as the ones for inactive robot in `recipe_electronic`.
For exemple robots using guns should have `gun_module` and laser turret has laser_canon in it s recipe, so either change the recipe to use the rifle or change  the broken robot to give the canon.

**Comment by NickGrease** [comment]
> Deconstruction recipe for broken robot should be the same as the ones for inactive robot

Not that I want to argue the point, or even care one way or the other, but based on language alone, I'd say broken != inactive. One implies damage (hence unrecoverable parts) and one implies intact parts just not active (hence all parts *should* be recoverable) I digress, if this is not the intended "meaning" of broken/inactive, then yes a change is needed.

**Comment by Fris0uman** [comment]
I'd need to double check the code but damage to the robot should carry over to deconstruction results.

Otherwise removing fragile component from the recipe is fine, but having different component makes no sense.

**Comment by NickGrease** [comment]
> Damage should carry over from robot to broken_robot

My apologies, I understand what you mean now; `broken_robot` should have the same disassembly components as robot/inactive_robot, and 'damage' will be taken into consideration for what parts are *actually* salvageable.

File: src/monstergenerator.cpp [hunk 35]
```diff
@@ -147,7 +147,9 @@ const std::map<std::string, m_flag> flag_map = {
     { "DRIPS_NAPALM", MF_DRIPS_NAPALM },
     { "DRIPS_GASOLINE", MF_DRIPS_GASOLINE },
     { "ELECTRIC_FIELD", MF_ELECTRIC_FIELD },
-    { "LOUDMOVES", MF_LOUDMOVES }
+    { "LOUDMOVES", MF_LOUDMOVES },
+    { "DROPS_AMMO", MF_DROPS_AMMO } //NEW
```
Comment (KorGgenT) [inline src/monstergenerator.cpp]:
```suggestion
    { "DROPS_AMMO", MF_DROPS_AMMO }
```
It won't be new forever. and the comment doesn't convey any useful informtation.

File: src/monstergenerator.cpp [hunk 36]
```diff
@@ -147,7 +147,9 @@ const std::map<std::string, m_flag> flag_map = {
...
+
```
Comment (KorGgenT) [inline src/monstergenerator.cpp]:
```suggestion
```

File: src/mtype.h [hunk 37]
```diff
@@ -164,7 +164,9 @@ enum m_flag : int {
     MF_ELECTRIC_FIELD,      // This monster is surrounded by an electrical field that ignites flammable liquids near it
     MF_LOUDMOVES,           // This monster makes move noises as if ~2 sizes louder, even if flying.
     MF_CAN_OPEN_DOORS,      // This monster can open doors.
-    MF_MAX                  // Sets the length of the flags - obviously must be LAST
+    MF_DROPS_AMMO,			// DROPS AMMO ON DEATH NEW!
```
Comment (KorGgenT) [inline src/mtype.h]:
```suggestion
    MF_DROPS_AMMO,			// Drops ammo on death
```

File: src/mtype.h [hunk 38]
```diff
@@ -164,7 +164,9 @@ enum m_flag : int {
...
+    MF_MAX                 // Sets the length of the flags - obviously must be LAST
+
```
Comment (KorGgenT) [inline src/mtype.h]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 39]
```diff
@@ -399,6 +399,32 @@
       [ [ "canister_empty", 1 ] ]
     ]
   },
+
```
Comment (KorGgenT) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
```

File: data/json/recipes/recipe_deconstruction.json [hunk 40]
```diff
@@ -421,6 +447,88 @@
...
+      [ [ "scrap", 1 ] ],
+      [ [ "sensor_module", 1 ] ],
+      [ [ "identification_module", 1 ] ],
+      [ [ "power_supply", 1 ] ],
+      [ [ "laser_rifle", 1 ] ],
+      [ [ "heavy_battery_cell", 1 ] ],
+      [ [ "scrap", 4 ] ]
+    ]
+  },
+
```
Comment (KorGgenT) [inline data/json/recipes/recipe_deconstruction.json]:
```suggestion
```

File: src/mondeath.cpp [hunk 41]
```diff
@@ -648,15 +648,25 @@ void mdeath::broken( monster &z )
...
+    //adds ammo drop	
+    const bool drops_ammo = z.type->has_flag( MF_DROPS_AMMO ); 
+    if( drops_ammo ) {
+    for( const auto &ammo_entry : z.type->starting_ammo ) {
+		if( z.ammo[ammo_entry.first] > 0 ) {
+			g->m.spawn_item( z.pos(), ammo_entry.first.c_str(), z.ammo[ammo_entry.first.c_str()], 1, calendar::turn );
+		}
+	}
+}
+    //end adds ammo drop
```
Comment (BevapDin) [inline src/mondeath.cpp]: This is not needed. If you want to give a piece of code some name, put it into a function.

File: src/mondeath.cpp (hunk 18 above)
Comment (misterprimus) [reply src/mondeath.cpp]: OK I did it can you check if it's in the PR?

**Comment by misterprimus** [comment]
> Decontruction recipe for broken robot should be the same as the ones for inactive robot in `recipe_electronic`.
> For exemple robots using guns should have `gun_module` and laser turret has laser_canon in it s recipe, so either change the recipe to use the rifle or change the broken robot to give the canon.

Yes, it should be consistent. I copied the components from the file you referenced.

**Comment by misterprimus** [comment]
> All references to volume, as I've just recently learned, should be strings in the form/conversion of 1 == 250 ml - so turrets ("volume": 120) should be `"volume": "30 L"` @anothersimulacrum can correct me if I'm wrong, but if 'xxxx ml' evenly divides by 1000, then use `L`, otherwise use `ml` (eg. "2500 ml" as opposed to "2.5 L")

Yeah, will keep in mind for future. Otherwise it would be easier to make the change _en masse_ in a future PR.

**Comment by anothersimulacrum** [comment]
Please make the change to the items in this PR, because you're introducing items here,

**Comment by anothersimulacrum** [comment]
Jenkins Rebuild

File: data/json/items/generic.json [hunk 42]
```diff
@@ -2910,5 +2910,90 @@
...
+    "id": "broken_turret",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it's laid limp on solid ground.  Could be gutted for parts.",
+    "price": 1000,
+    "material": [ "steel", "plastic" ],
+    "weight": 62650,
+    "volume": 120,
```
Comment (anothersimulacrum) [inline data/json/items/generic.json]:
```suggestion
    "volume": "30 L",
```

File: data/json/items/generic.json [hunk 43]
```diff
@@ -2910,5 +2910,90 @@
...
+    "id": "broken_turret_rifle",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken milspec turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it's laid limp on solid ground.  Could be gutted for parts.",
+    "price": 1000,
+    "material": [ "steel", "plastic" ],
+    "weight": 62650,
+    "volume": 120,
```
Comment (anothersimulacrum) [inline data/json/items/generic.json]:
```suggestion
    "volume": "30 L",
```

File: data/json/items/generic.json [hunk 44]
```diff
@@ -2910,5 +2910,90 @@
...
+    "id": "broken_turret_bmg",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken antimateriel turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it's laid limp on solid ground.  Could be gutted for parts.",
+    "price": 1000,
+    "material": [ "steel", "plastic" ],
+    "weight": 62650,
+    "volume": 120,
```
Comment (anothersimulacrum) [inline data/json/items/generic.json]:
```suggestion
    "volume": "30 L",
```

File: data/json/items/generic.json [hunk 45]
```diff
@@ -2910,5 +2910,90 @@
...
+    "id": "broken_laserturret",
+    "symbol": ",",
+    "color": "green",
+    "name": "broken laser turret",
+    "category": "other",
+    "description": "A broken turret.  Much less threatening now that it's laid limp on solid ground.  Could be gutted for parts.",
+    "price": 1000,
+    "material": [ "steel", "plastic" ],
+    "weight": 62650,
+    "volume": 120,
```
Comment (anothersimulacrum) [inline data/json/items/generic.json]:
```suggestion
    "volume": "30 L",
```

File: src/mondeath.cpp [hunk 46]
```diff
@@ -654,6 +653,18 @@ void mdeath::broken( monster &z )
 
     g->m.add_item_or_charges( z.pos(), broken_mon );
 
+
+    // adds ammo drop
+    const bool drops_ammo = z.type->has_flag( MF_DROPS_AMMO );
```
Comment (BevapDin) [inline src/mondeath.cpp]:
```suggestion
    if( z.type->has_flag( MF_DROPS_AMMO ) ) {
```
No need for a separate variable.

**Pull request merged**